packages/discovery-react-components/src/components/DocumentPreview/__tests__/DocumentPreview.test.tsx

```diff
@@ -28,91 +74,265 @@ describe('DocumentPreview', () => {
-      element: null,
-      elementType: null
-    };
-    const results = {
-      matching_results: 1,
-      results: [selectedResult]
-    };
-
-    let getByText: NonNullable<any>;
-    act(() => {
-      ({ getByText } = render(
-        <DiscoverySearch
-          searchClient={searchClient}
-          projectId={'PROJECT_ID'}
-          overrideSearchResults={results}
-          overrideSelectedResult={selectedResult}
-        >
-          <DocumentPreview />
-        </DiscoverySearch>
-      ));
+
+    it('renders an overwritten "body" field', async () => {
+      context = {
+        componentSettings: {
+          fields_shown: {
+            body: {
+              field: 'body_field'
+            }
+          }
+        }
+      };
+      act(() => {
+        ({ getByText } = render(
+          wrapWithContext(<DocumentPreview document={htmlTextDoc} />, api, context)
+        ));
+      });
+
+      getByText('Display text from the specified "body" field.');
     });
 
-    getByText('On 22 December 2008 ART EFFECTS LIMITED', { exact: false });
+    it('renders with data from selected result', async () => {
+      const authenticator = new NoAuthAuthenticator();
+      const searchClient = new DiscoveryV2({
+        ur: 'http://mock:3000/api',
+        version: '2019-01-01',
+        authenticator
+      });
+      const dummyResponse = {
+        result: {},
+        status: 200,
+        statusText: 'OK',
+        headers: {}
+      };
+      jest
+        .spyOn(searchClient, 'listCollections')
+        .mockImplementation(() => Promise.resolve(dummyResponse));
+      jest
+        .spyOn(searchClient, 'getComponentSettings')
+        .mockImplementation(() => Promise.resolve(dummyResponse));
+
+      const selectedResult = {
+        document: omit(pdfDoc, 'extracted_metadata.text_mappings'),
+        element: null,
+        elementType: null
+      };
+      const results = {
+        matching_results: 1,
+        results: [selectedResult]
+      };
+
+      act(() => {
+        ({ getByText } = render(
+          <DiscoverySearch
+            searchClient={searchClient}
+            projectId={'PROJECT_ID'}
+            overrideSearchResults={results}
+            overrideSelectedResult={selectedResult}
+          >
+            <DocumentPreview />
+          </DiscoverySearch>
+        ));
+      });
+      getByText('On 22 December 2008 ART EFFECTS LIMITED', { exact: false });
+    });
   });
 
-  it('renders document with html-like text', () => {
-    const htmlTextDoc = {
-      id: '1234567890',
-      extracted_metadata: {
-        filename: 'i_am_a_file'
-      },
-      text: ['Example text <text> <username> <password> more text afterwards']
-    };
+  // All of these currently fail since DocumentPreview defaults to HtmlView if a HTML field is present.
+  // Also fails because SimpleDocument will not show a JSON document without a passage highlight.
+  describe.skip('JSON files', () => {
```

Comment:
Bring `jsonDoc` down to right before this `describe`. Makes it easier to read the tests.